src/main/java/seedu/address/commons/util/StringUtil.java

```diff
@@ -65,4 +65,30 @@ public static boolean isNonZeroUnsignedInteger(String s) {
             return false;
         }
     }
+
+    /**
+     * Returns true if {@code s} represents a non-zero dollar amount with at most 2 decimal places,
+     * e.g. 12.20, 5, 0.01.
+     * Will return false for any other non-null String input, e.g., "-2.00", "0".
+     *
+     * @throws NullPointerException if {@code s} is null.
+     */
+    public static boolean isValidDollarAmount(String s) {
+        requireNonNull(s);
+
+        String[] dollarsAndCents = s.split("\\."); // split by the dot '.'
+        if (dollarsAndCents.length > 1 && dollarsAndCents[1].length() > 2) {
+            return false;
+        }
+
+        try {
+            double value = Double.parseDouble(s);
+            if (value * 100.0 > (double) Integer.MAX_VALUE) { // value in cents is greater than max int value
```

Comment:
Maybe we can use the method `dollarToCents` from `FineUtil`

Reply:
That method returns an int but the value here can be larger than max int value which is what I'm testing